streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -281,48 +275,63 @@
     <V1, V2> KStream<K, V2> leftJoin(KTable<K, V1> table, ValueJoiner<V, V1, V2> joiner);
 
     /**
-     * Aggregate values of this stream by key on a window basis.
+     * Combine values of this stream by key on a window basis into a new instance of windowed {@link KTable}.
```

Comment:
I agree. Though I'd rather do it in a separate PR since I want to describe this notion in web docs / java docs as a whole. 

This may also be modified in the future if we see a better solution in terms of API for specifying windowed result.